src/commands/functions-config-export.ts

```diff
@@ -0,0 +1,383 @@
+    errMsg,
+  };
+}
+
+async function promptForPrefix(errMsg: string): Promise<string> {
+  logWarning("The following configs keys could not be exported as environment variables:\n");
+  logWarning(errMsg);
+  return await promptOnce(
+    {
+      type: "input",
+      name: "prefix",
+      default: "CONFIG_",
+      message: "Enter a PREFIX to rename invalid environment variable keys:",
+    },
+    { nonInteractive: false }
+  );
+}
+
+function escape(s: string): string {
+  // Escape newlines and tabs
+  let result = s
+    .replace("\n", "\\n")
+    .replace("\r", "\\r")
+    .replace("\t", "\\t")
+    .replace("\v", "\\v");
+  // Escape other escape characters like ' and ".
+  result = result.replace(/(['"])/g, "\\$1");
+  return result;
+}
+
+/**
+ * Convert env var mappings to string in dotenv format.
+ */
+export function toDotenvFormat(envs: EnvMap[], header = ""): string {
+  const lines = envs.map(({ newKey, value }) => `${newKey}="${escape(value)}"`);
+  const maxLineLen = Math.max(...lines.map((l) => l.length));
+  return (
+    `${header}\n` +
+    lines.map((line, idx) => `${line.padEnd(maxLineLen)} # from ${envs[idx].origKey}`).join("\n")
+  );
+}
+
+function writeEnvs(
+  basePath: string,
+  header: string,
+  projects: Project[],
+  envs: EnvMap[][]
+): string[] {
+  return envs.map((env, idx) => {
+    const project = projects[idx];
+    const dotenv = toDotenvFormat(env, header);
+    const ext = project.alias ?? project.projectId;
+    const filename = ext ? `.env.${ext}` : ".env";
+    const filePath = path.join(basePath, filename);
+
+    fs.writeFileSync(filePath, dotenv);
+    return filePath;
+  });
+}
+
+function writeEmptyEnvs(basePath: string, header: string, projects: Project[]): string[] {
+  return writeEnvs(
+    basePath,
+    header,
+    projects,
+    projects.map(() => [])
+  );
+}
+
+async function copyFilesToDir(srcFiles: string[], destDir: string): Promise<string[]> {
+  const destFiles = [];
+  for (const file of srcFiles) {
+    const targetFile = path.join(destDir, path.basename(file));
+    if (fs.existsSync(targetFile)) {
+      const overwrite = await promptOnce(
+        {
+          type: "confirm",
+          name: "overwrite",
+          default: true,
+          message: `${targetFile} already exists. Overwrite file?`,
+        },
+        { nonInteractive: false }
+      );
+      if (!overwrite) {
+        logBullet(`Skipping ${targetFile}`);
+        continue;
+      }
+    }
+    fs.copyFileSync(file, targetFile);
+    destFiles.push(targetFile);
+  }
+  return destFiles.filter((f) => f.length > 0);
+}
+
+export default new Command("functions:config:export")
+  .description("Export environment config as environment variables in dotenv format")
+  .before(requirePermissions, [
+    "runtimeconfig.configs.list",
+    "runtimeconfig.configs.get",
+    "runtimeconfig.variables.list",
+    "runtimeconfig.variables.get",
+  ])
+  .action(async (options: any) => {
```

Comment:
Might want to just throw if `options.nonInteractive` is true since this command won't work in non-interactive mode.